gslib/tests/test_wrapped_credentials.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for wrapped_credentials.py."""
+
+import datetime
+import json
+import httplib2
+
+from google.auth import credentials
+from google.auth import external_account
+from google.auth import identity_pool
+from gslib.tests import testcase
+from gslib.utils.wrapped_credentials import WrappedCredentials
+import oauth2client
+
+from six import add_move, MovedModule
+
+add_move(MovedModule("mock", "mock", "unittest.mock"))
+from six.moves import mock
+
+ACCESS_TOKEN = "foo"
+CONTENT = "content"
+RESPONSE = httplib2.Response({
+    "content-type": "text/plain",
+    "status": "200",
+    "content-length": len(CONTENT),
+})
+
+
+class MockCredentials(external_account.Credentials):
+  refresh = mock.Mock()
```

Comment:
Why not also declare this in `__init__`?
```
self.refresh = mock.Mock(side_effect=side_effect)
```

Reply:
Done